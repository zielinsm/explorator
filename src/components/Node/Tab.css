.Tab__row {
    padding-bottom: 1em;
}

.Tab__field {
    word-wrap: break-word;
    white-space: pre-wrap;
    max-height: 100px;
    overflow: auto;
}
