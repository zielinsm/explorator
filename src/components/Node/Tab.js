import React from "react";
import { Grid, Header } from "semantic-ui-react";

import "./Tab.css";

const convertArray = function convertArrayToAString(element) {
  if (Array.isArray(element)) {
    let formatted = "";
    element.forEach(value => {
      formatted += `${value} \n`;
    });
    return formatted;
  } else {
    return element;
  }
};

const elementCheck = function checkElementForQuirks(value) {
  if (value === undefined || value === null) {
    return "Not defined";
  } else {
    return convertArray(value);
  }
};

function Tab(props) {
  const { data } = props;

  const renderRow = function renderDefaultRow(data, index) {
    const value = elementCheck(data.value);
    const { label } = data;

    return (
      <Grid.Row className="Tab__row" key={index}>
        <Header sub>{label}</Header>
        <span className="Tab__field">{value}</span>
      </Grid.Row>
    );
  };

  return data.map((element, index) => renderRow(element, index));
}

export default Tab;
