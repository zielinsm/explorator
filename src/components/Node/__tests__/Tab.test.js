import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import Tab from "../Tab";

const mockData = [
  { label: "mockLabel", value: "mockValue" },
  { label: "someLabel", value: "someLabel" }
];

describe("Tab component:", () => {
  const component = <Tab data={mockData} />;

  it("renders data passed as props", () => {
    const wrapper = mount(component);
    expect(wrapper.find(".Tab__field")).toHaveLength(2);
  });

  it("snapshot matches", () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
