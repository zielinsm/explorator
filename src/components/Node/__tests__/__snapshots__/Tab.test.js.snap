// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab component: snapshot matches 1`] = `
Array [
  <div
    className="row Tab__row"
>
    <div
        className="ui sub header"
    >
        mockLabel
    </div>
    <span
        className="Tab__field"
    >
        mockValue
    </span>
</div>,
  <div
    className="row Tab__row"
>
    <div
        className="ui sub header"
    >
        someLabel
    </div>
    <span
        className="Tab__field"
    >
        someLabel
    </span>
</div>,
]
`;
